subsys/retention/Kconfig

```diff
@@ -55,6 +55,46 @@ config RETENTION_BOOT_MODE
 	  byte must be created and set as the "zephyr,boot-mode" chosen node
 	  via device tree.
 
+menuconfig RETENTION_BOOTLOADER_CONFIG
+	bool "Bootloader configuration"
+	depends on SETTINGS
+	help
+	  Adds a bootloader configuration sharing system which allows for
+	  retreiving information from the bootloader when data sharing is
+	  enabled via the settings system with the "blinfo" prefix.
+
+if RETENTION_BOOTLOADER_CONFIG
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_BOOTLOADER_CONFIG := zephyr,bootloader-config
+
+config RETENTION_BOOTLOADER_CONFIG_TYPE_MCUBOOT
+	bool "MCUboot"
+	depends on !MCUBOOT && BOOTLOADER_MCUBOOT
+	depends on $(dt_chosen_enabled,$(DT_CHOSEN_BOOTLOADER_CONFIG))
+	select MCUBOOT_BOOTUTIL_LIB
+	help
+	  Adds a bootloader configuration sharing system for MCUboot and
+	  applications which allows applications to read the configurion of
+	  MCUboot and the running image. This can be used by applications so
+	  that they know how to e.g. handle firmware updates and place them
```

Comment:
We will need to modify the `active_image` and `active_slot` functions to be aware of that.